synapse/lib/layer.py

```diff
@@ -1101,27 +1117,27 @@
     def __repr__(self):
         return f'Layer ({self.__class__.__name__}): {self.iden}'
 
-    async def __anit__(self, layrinfo, dirn, nexsroot=None, allow_upstream=True, mapasync=True, maxreplaylog=10000):
+    async def __anit__(self, core, layrinfo):
 
-        self.nexsroot = nexsroot
+        self.core = core
         self.layrinfo = layrinfo
-        self.allow_upstream = allow_upstream
 
         self.addoffs = None  # The nexus log index where I was created
         self.deloffs = None  # The nexus log index where I was deleted
         self.isdeleted = False
 
         self.iden = layrinfo.get('iden')
-        await s_nexus.Pusher.__anit__(self, self.iden, nexsroot=nexsroot)
+        await s_nexus.Pusher.__anit__(self, self.iden, nexsroot=core.nexsroot)
 
-        self.dirn = dirn
+        self.dirn = s_common.gendir(core.dirn, 'layers', self.iden)
         self.readonly = layrinfo.get('readonly')
 
         self.lockmemory = self.layrinfo.get('lockmemory')
         self.growsize = self.layrinfo.get('growsize')
         self.logedits = self.layrinfo.get('logedits')
-        self.mapasync = mapasync
-        self.maxreplaylog = maxreplaylog
+
+        self.mapasync = core.conf.get('layer:lmdb:map_async')
```

Comment:
Might there be an advantage to keeping some sane defaults local to Layer so that a caller doesn't have to keep up with conf changes?  I'm thinking of the datanode for example, but there might be others.